lms/services/h_api.py

```diff
@@ -159,43 +72,33 @@ def _api_request(self, method, path, data=None, headers=None, body=None):
         :param path: the h API path to post to, relative to
                      ``settings["h_api_url_private"]``, for example:
                      ``"users"`` or ``"groups/<GROUPID>/members/<USERID>"``
-        :param data: the data to post as JSON in the request body
         :param body: the body to send as a string (without modification)
         :param headers: extra headers to pass with the request
 
-        :raise HAPINotFoundError: If the request fails with 404
         :raise HAPIError: if the request fails for any other reason
         :return: the response from the h API
         :rtype: requests.Response
         """
         headers = headers or {}
         headers["Hypothesis-Application"] = "lms"
 
-        request_args = {"headers": headers}
-
+        request_args = {}
```

Comment:
It was a bit weird to set headers like this, when we set everything else directly below. The reason to do this is when we don't know if the arg will be set at all (like "data" below). We always set headers so I've used a named parameter to the call to request below.